.devcontainer/Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM mcr.microsoft.com/dotnet/core/sdk:3.1
+
+RUN apt update && apt -y install --no-install-recommends --no-install-suggests \
+        bash-completion \
+        less \
+        nano \
+ && apt -y autoremove \
+ && apt -y clean \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN groupadd -g 1000 dev \
+ && useradd -g 1000 -u 1000 -m -s /bin/bash dev
+
+USER dev
```

Comment:
The commands in all capitals are normal `Dockerfile` commands. When building an image, you start from a specified base image `FROM mcr.microsoft.com/dotnet/core/sdk:3.1`, then each command runs in sequence to build the final image.

The base image name is taken from [Docker hub][2].

The commands _inside_ the `RUN` are normal unix utilities, since the base image we're using is a `Debian` distro.

The `apt` install command is [considered a best practice][1] to keep the image as small as possible.

[1]: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run
[2]: https://hub.docker.com/_/microsoft-dotnet-sdk